### PR TITLE
Fix: Cleanup XAML codes

### PR DESCRIPTION
**Pull request by onein528**
### Description

Adding `Mode=OneWay` in DataTemplates are not needed and also there should not be them. This addition have caused a lot of WMC1506 warnings.

### Motive and Context

- Reduce `WMC1506: OneWay bindings require at least one of their steps to support raising notifications when their value changes`
- Add lines and comments to improve readability

Note
- No UI Element changes.
- CreateShortcutDialog and DecompressArchiveDialog has unwrapping changes; unneeded wrapping by Border or Grid.
- Gird.HorizontalContent is Stretch by default. so removed.

### Validation

- [x] Built and ran the app
- [x] Tested the changes for accessibility

### CODE REVIEW

**Comment by onein528** [comment]
Tests went wrong.

File: src/Files.App/Dialogs/AddItemDialog.xaml [hunk 1]
```diff
@@ -22,55 +18,64 @@
...
-	<Grid RowSpacing="25">
+	<Grid x:Name="RootGrid" RowSpacing="24">
 		<Grid.RowDefinitions>
 			<RowDefinition Height="Auto" />
 			<RowDefinition Height="*" />
 		</Grid.RowDefinitions>
-		<Grid x:Name="SubtitleArea">
+
+		<!--  Description  -->
+		<Grid x:Name="SubtitleArea" Grid.Row="0">
```
Comment (yaira2) [inline src/Files.App/Dialogs/AddItemDialog.xaml:28]: Do we even need this grid?

File: src/Files.App/Dialogs/CredentialDialog.xaml [hunk 2]
```diff
@@ -22,31 +22,42 @@
 		</ResourceDictionary>
 	</ContentDialog.Resources>
 
-	<Grid RowSpacing="25">
+	<Grid RowSpacing="24">
```
Comment (yaira2) [inline src/Files.App/Dialogs/CredentialDialog.xaml:25]: Thanks!

Comment (onein528) [reply src/Files.App/Dialogs/CredentialDialog.xaml:25]: There're many:) 25 changed to 24, 50 changed to 50.

File: src/Files.App/Dialogs/AddItemDialog.xaml (hunk 1 above)
Comment (onein528) [reply src/Files.App/Dialogs/AddItemDialog.xaml:28]: Removed

**Pull request closed without merging**